packages/demo/backend/src/controllers/lend.ts

```diff
@@ -118,79 +120,60 @@ export class LendController {
+      const validation = await validateRequest(c, OpenPositionRequestSchema)
       if (!validation.success) return validation.response
 
       const {
-        body: { walletId, amount, tokenAddress, chainId },
+        body: { walletId, amount, tokenAddress, chainId, vaultAddress },
       } = validation.data
       const auth = c.get('auth') as AuthContext | undefined
 
+      let hash: string
+
       // TODO (https://github.com/ethereum-optimism/verbs/issues/124): enforce auth and clean
       // up this route.
-      if (auth && auth.userId) {
-        const lendTransaction = await lendService.depositWithUserWallet(
-          auth.userId,
-          amount,
-          tokenAddress as Address,
-          chainId as SupportedChainId,
-        )
-
-        const result = await lendService.executeLendTransactionWithUserWallet(
-          auth.userId,
-          lendTransaction,
-          chainId as SupportedChainId,
-        )
+      const params = {
+        amount,
+        asset: {
+          tokenAddress: tokenAddress as Address,
+          chainId: chainId as SupportedChainId,
+        },
+        marketId: {
+          address: vaultAddress as Address,
+          chainId: chainId as SupportedChainId,
+        },
+      }
 
-        return c.json({
-          transaction: {
-            blockExplorerUrl: result.blockExplorerUrl,
-            hash: result.hash,
-            amount: result.amount.toString(),
-            asset: result.asset,
-            marketId: result.marketId,
-            apy: result.apy,
-            timestamp: result.timestamp,
-            slippage: result.slippage,
-            transactionData: serializeBigInt(result.transactionData),
-          },
-        })
+      // Use userId if authenticated, otherwise use walletId
+      if (auth && auth.userId) {
+        hash = await lendService.openPosition(auth.userId, params, true)
+      } else {
+        hash = await lendService.openPosition(walletId, params, false)
       }
 
-      const lendTransaction = await lendService.deposit(
-        walletId,
-        amount,
-        tokenAddress as Address,
-        chainId as SupportedChainId,
-      )
-      const result = await lendService.executeLendTransaction(
-        walletId,
-        lendTransaction,
+      const blockExplorerUrl = await lendService.getBlockExplorerUrl(
         chainId as SupportedChainId,
       )
 
       return c.json({
         transaction: {
-          blockExplorerUrl: result.blockExplorerUrl,
-          hash: result.hash,
-          amount: result.amount.toString(),
-          asset: result.asset,
-          marketId: result.marketId,
-          apy: result.apy,
-          timestamp: result.timestamp,
-          slippage: result.slippage,
-          transactionData: serializeBigInt(result.transactionData),
+          hash,
+          blockExplorerUrl,
+          amount,
+          tokenAddress,
+          chainId,
+          vaultAddress,
         },
```

Comment:
ℹ️ These are all the frontend needs, but I'll improve this return type in [this issue](https://github.com/ethereum-optimism/verbs/issues/72)